packages/ragbits-core/src/ragbits/core/utils/decorators.py

```diff
@@ -0,0 +1,55 @@
+# pylint: disable=missing-function-docstring,missing-return-doc
+
+import asyncio
+from functools import wraps
+from importlib.util import find_spec
+from typing import Callable, ParamSpec, TypeVar
+
+_P = ParamSpec("_P")
+_T = TypeVar("_T")
+
+
+def requires_dependencies(
```

Comment:
I'd add some tests for this func. It should be fairly easy: we can pass `pytest` or `ragbits` as deps as they are always present, and some made up deps such as `micpst_super_secret_lib` to raise `ImportError`

Reply:
done, thx